server/controllers/chartsController/helpers.test.js

```diff
@@ -739,4 +836,169 @@ describe('chartsController - helpers', () => {
+        inclusionCriteria: [
+          {
+            $match: {
+              assessment: INCLUSION_EXCLUSION_CRITERIA_FORM,
+              study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+            },
+          },
+          {
+            $project: {
+              ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+              inclusionCriteriaCollection: MONGO_COLLECTION_STRING,
+            },
+          },
+        ],
+      })
+    })
+
+    it('returns an object only for the requested filters', () => {
+      expect(
+        helpers.buildFacetForFilters({
+          isSexAtBirthFilterActive: true,
+          isInclusionCriteriaFilterActive: undefined,
+          userAccess,
+        })
+      ).toEqual({
+        socioDemographics: [
+          {
+            $match: {
+              assessment: SOCIODEMOGRAPHICS_FORM,
+              study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+            },
+          },
+          {
+            $project: {
+              ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+              socioDemographicsCollection: MONGO_COLLECTION_STRING,
+            },
+          },
+        ],
+      })
+    })
+  })
+
+  describe(helpers.intersectSubjectsFromFilters, () => {
+    it('adds subjects to the resultsFromFiltersMap', () => {
+      const resultsFromFiltersMap = new Map()
+      const requestedFilters = {
+        socioDemographics: [
+          {
+            study: 'YA',
+            subject: 'YA01508',
+            socioDemographicsCollection:
+              '99fc2c55c6718fcd8eadf9244a24d2f68aae77c6fc591d5e2a8e67bab906446f',
+          },
+        ],
+        inclusionCriteria: [
+          {
+            study: 'YA',
+            subject: 'YA01508',
+            inclusionCriteriaCollection:
+              '062419c48a8b6634e696b9dc0674fb87c34aea44751d1eba107feb0fc7325d71',
+          },
+          {
+            study: 'CA',
+            subject: 'CA00063',
+            inclusionCriteriaCollection: 'CA00063filter',
+          },
+          {
+            study: 'LA',
+            subject: 'LA00028',
+            inclusionCriteriaCollection: 'la00028filter',
+          },
+          {
+            study: 'YA',
+            subject: 'YA00015',
+            inclusionCriteriaCollection: 'YA00015Filter',
+          },
+        ],
+      }
+      helpers.intersectSubjectsFromFilters(
+        requestedFilters,
+        resultsFromFiltersMap
+      )
+      expect(resultsFromFiltersMap.size).toEqual(1)
+    })
+
+    it('updates subject data if subject has been added to map', () => {
+      const resultsFromFiltersMap = new Map()
+      resultsFromFiltersMap.set('YA01508', {
+        subject: 'YA01508',
+        socioDemographicsCollection:
+          '99fc2c55c6718fcd8eadf9244a24d2f68aae77c6fc591d5e2a8e67bab906446f',
+      })
+      const requestedFilters = {
+        socioDemographics: [
+          {
+            study: 'YA',
+            subject: 'YA01508',
+            socioDemographicsCollection:
```

Comment:
is this key coming from the database or are we naming it `socioDemographicsCollection`? If so, what do you think about naming it `collection` instead? Same goes for `inclusionCriteriaCollection`. That way the shape of the data is consistent between these filters.

Reply:
I had to rename the collections from collection because after we intersect the subject we have to query the different assessments for each subjects for each requested filter. If the stays the same then it would replace the value if a filter from a different assessment is requested